core/browser_buffer.py

```diff
@@ -90,3 +90,9 @@ def scroll_to_begin(self):
 
     def scroll_to_bottom(self):
         self.eval_js("window.scrollBy(0, document.body.scrollHeight)")
+
+    def set_bookmark(self):
+        url = self.buffer_widget.web_page.executeJavaScript("window.location.href;")
+        self.buffer_widget.set_emacs_var.emit("eaf--browser-current-url", url)
+
```

Comment:
Check patch https://github.com/manateelazycat/emacs-application-framework/commit/465c3f3ec4a631932477e5e3e9f1bb77c62d9ba8

Change set_bookmark to get_bookmark:

def get_bookmark(self):
    return self.buffer_widget.web_page.executeJavaScript("window.location.href;")

Reply:
Nice, thanks, this dbus stuff is still black magic to me ;)

Reply:
Default, execute_function just execute function, call_function return Python function result to Elisp side through DBus IPC, then you don't need set_emacs_var.emit set temp elisp variable everywhere. ;)